tycho-indexer/src/extractor/dynamic_contract_indexer.rs

```diff
@@ -0,0 +1,348 @@
+    pub async fn initialize(&mut self) {
+        let entrypoint_filter = EntryPointFilter::new(Some(self.protocol.clone()));
+
+        let entrypoints = self
+            .entrypoint_gw
+            .get_entry_points_with_data(entrypoint_filter)
+            .await
+            .unwrap();
+
+        for entrypoint in entrypoints {
+            self.protocol_entrypoints
+                .insert(entrypoint.clone());
+
+            let traced_entrypoints = self
+                .entrypoint_gw
+                .get_traced_entry_point(entrypoint)
+                .await
+                .unwrap();
+            if let Some(traced_entrypoint) = traced_entrypoints {
+                self.entrypoint_results
+                    .insert(entrypoint, traced_entrypoint);
+            }
+        }
+    }
+
+    pub async fn process_block_update(
+        &mut self,
+        new_entrypoints: &HashMap<ComponentId, Vec<EntryPointWithData>>,
+        changes: &Vec<TransactionAccountUpdates>,
+        block: &Block,
+    ) -> DCIBlockUpdate {
+        let mut result_deltas: HashMap<Transaction, Vec<AccountDelta>> = HashMap::new();
+        let mut result_components: HashMap<ComponentId, Vec<Address>> = HashMap::new();
+
+        // Registers new Entrypoints sent by the extractor to the map of Component -> Entrypoints
+        for (component_id, entrypoints) in new_entrypoints {
+            for entrypoint in entrypoints {
+                self.protocol_entrypoints
+                    .get(&component_id)
+                    .get_or_insert_default()
+                    .push(entrypoint.clone());
+            }
+        }
+
+        // Select for analysis the newly detected EntryPointsWithData that haven't been analyzed
+        // yet. This filter prevents us from re-analyzing entrypoints that have already been
+        // analyzed, which can be a case if all the components have the same entrypoint.
+        let mut entrypoints_to_analyze = self
+            .protocol_entrypoints
+            .values()
+            .flatten()
+            .filter(|entrypoint| {
+                !self
+                    .entrypoint_results
+                    .contains_key(entrypoint)
+            })
+            .cloned()
+            .collect::<HashSet<_>>();
+
+        // Use block storage changes to detect retriggered entrypoints
+        let retriggered_entrypoints = self.detect_retriggers(&changes);
+
+        // Update the entrypoint results with the retriggered entrypoints
+        entrypoints_to_analyze.extend(retriggered_entrypoints);
+        debug!("Will analyze {:?} entrypoints", entrypoints_to_analyze.len());
+
+        // Trace analyze all the entrypoints
+        let entrypoints_to_analyze: Vec<EntryPointWithData> = entrypoints_to_analyze.into();
+
+        let traced_entry_points = self
+            .tracer
+            .trace(block.hash.clone(), entrypoints_to_analyze)
+            .await
+            .expect("Failed to trace entrypoints");
```

Comment:
So this panics if one EP fails to analyze? So then by design I think that `protocol_entrypoints` should be always empty?